base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ActionsViewTable.java

```diff
@@ -57,16 +67,55 @@ public ActionsViewTable(Composite parent, int style, int tableStyle, ScriptGener
         super(parent, style, tableStyle | SWT.BORDER);
         this.scriptGeneratorViewModel = scriptGeneratorViewModel;
         initialise();
-        
         scriptGeneratorViewModel.addActionParamPropertyListener(this);
+		TableViewerFocusCellManager focusCellManager = new TableViewerFocusCellManager(viewer, new FocusCellOwnerDrawHighlighter(viewer), new CellNavigationStrategy());
+		ColumnViewerEditorActivationStrategy activationStrategy = createEditorActivationStrategy(viewer);
+		TableViewerEditor.create(viewer, focusCellManager, activationStrategy, 
+				ColumnViewerEditor.TABBING_HORIZONTAL 
+				| ColumnViewerEditor.TABBING_MOVE_TO_ROW_NEIGHBOR
+				| ColumnViewerEditor.TABBING_CYCLE_IN_VIEWER
+				| ColumnViewerEditor.TABBING_VERTICAL);
     }
+    
+    /**
+     * Strategy for editing cell.
+     * @param tableViewer our table viewer.
+     * @return Editor Activation strategy.
+     */
+	private ColumnViewerEditorActivationStrategy createEditorActivationStrategy(TableViewer tableViewer) {
+		return new ColumnViewerEditorActivationStrategy(tableViewer) {
+			protected boolean isEditorActivationEvent(ColumnViewerEditorActivationEvent event) {
+				boolean retVal = true;
+				if (event.eventType == ColumnViewerEditorActivationEvent.TRAVERSAL) { 
+ 					ColumnViewerEditor editor = this.getViewer().getColumnViewerEditor();
+ 					ViewerCell nextCell = editor.getFocusCell().getNeighbor(ViewerCell.RIGHT, false);
+					int currentlyFocusedColumn = editor.getFocusCell().getColumnIndex() + 1;
```

Comment:
Given that the reason this variable is a 1 based index and not a 0 base index is because we want to go to a new row when we are on the penultime column, as the last column is a non editable validity column, this should be made clear in the code.

Renaming currentlyFocusedColumn and leaving a comment would suffice. You could also subtract 2 instead of 1 in the if statement, but i do not think that would be as clear.

Reply:
I think the name currentlyFocusedColumn represents correctly what the variable does. It holds the column number of a cell that is currently focused.